specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,225 @@
+
+`TARGET_NUMBER_OF_PEERS` should be tuned upward in the event of failed sampling.
+
+*Note*: while high-capacity and super-full nodes are high value with respect to satisfying sampling requirements, a node should maintain a distribution across node capacities as to not centralize the p2p graph too much (in the extreme becomes hub/spoke) and to distribute sampling load better across all nodes.
+
+*Note*: A DHT-based peer discovery mechanism is expected to be utilized in the above. The beacon-chain network currently utilizes discv5 in a similar method as described for finding peers of particular distributions of attestation subnets. Additional peer discovery methods are valuable to integrate (e.g., latent peer discovery via libp2p gossipsub) to add a defense in breadth against one of the discovery methods being attacked.
+
+## Extended data
+
+In this construction, we extend the blobs using a one-dimensional erasure coding extension. The matrix comprises maximum `MAX_BLOBS_PER_BLOCK` rows and fixed `NUMBER_OF_COLUMNS` columns, with each row containing a `Blob` and its corresponding extension.
+
+## Column gossip
+
+### Parameters
+
+For each column -- use `data_column_sidecar_{subnet_id}` subnets, where each column index maps to the `subnet_id`. The sidecars can be computed with `get_data_column_sidecars(signed_block: SignedBeaconBlock, blobs: Sequence[Blob])` helper.
+
+To custody a particular column, a node joins the respective gossip subnet. Verifiable samples from their respective column are gossiped on the assigned subnet.
+
+### Reconstruction and cross-seeding
+
+If the node obtains 50%+ of all the columns, they can reconstruct the full data matrix via `recover_samples_impl` helper.
+
+If a node fails to sample a peer or fails to get a column on the column subnet, a node can utilize the Req/Resp message to query the missing column from other peers.
+
+Once the node obtain the column, the node should send the missing columns to the column subnets.
+
+*Note*: A node always maintains a matrix view of the rows and columns they are following, able to cross-reference and cross-seed in either direction.
+
+*Note*: There are timing considerations to analyze -- at what point does a node consider samples missing and choose to reconstruct and cross-seed.
+
+*Note*: There may be anti-DoS and quality-of-service considerations around how to send samples and consider samples -- is each individual sample a message or are they sent in aggregate forms.
+
+## Peer sampling
+
+At each slot, a node makes (locally randomly determined) `SAMPLES_PER_SLOT` queries for samples from their peers via `DataColumnSidecarByRoot` request. A node utilizes `get_custody_lines` helper to determine which peer(s) to request from. If a node has enough good/honest peers across all rows and columns, this has a high chance of success.
```

Comment:
```suggestion
At each slot, a node makes (locally randomly determined) `SAMPLES_PER_SLOT` queries for samples from their peers via `DataColumnSidecarByRoot` request. A node utilizes `get_custody_lines` helper to determine which peer(s) to request from. If a node has enough good/honest peers across all rows and columns, this has a high chance of success.
```
from all peers on a particular line or from a randomly sampled subset? (if more than 1 peer available)

Reply:
similar to sync, etc, requesting from peers is a local load-balancing and other optimization. I don't think the spec should dictate

I need to get some random set of columns, I look at my peer-set who is distributed across a diverse set of column custody, I make the requests to the peers that are supposed to have what I need based on local strategy/hueristic

Reply:
understood, but expecting nodes to `randomly` sample peers (from the column subnet) for lets say each column would put stronger guarantees on the DA proofs? else there is a chance to endup on  a hub/spoke model if nodes hook onto the super nodes.

Reply:
The failure here ends up being network liveness rather than safety. If there is an over reliance on a hub and it fails.

There is certainly a chance, as peers attempt to balance diversity of peer with guarantees that they'll respond. I do think that it is a clients job to hueristically balance this to ensure that if the hub defaults, that they have adequate diversity to fill in the gaps.

this is similar to attnet peer balancing -- peers of high attnets are very valuable but an over-reliance can create a brittle network.

Fortunately, the failure of an overly hub-spoked network is a temporary liveness, rather than safety failure

Reply:
With the current numbers we have, a given column is sampled often by a given node, on average every `NUMBER_OF_COLUMNS / SAMPLES_PER_SLOT` slots. That's a few minutes.

If nodes do some rotation or randomization of where to sample from, or e.g. ask two nodes for the same sample (maybe not for all samples, only where they miss liveness info on peers for the given column), they can easily keep a list of active responding peers for each column.

Speed matters, and we are already OK with duplicates (at least in the GossipSub part), and from the safety perspective it does not matter where the sample comes from. I would use this duplicate sampling strategy, but this is an implementation decision.

I think for the EIP a SHOULD sounds good. Something like `a node SHOULD maintain a diverse set of peers for each column by verifying responsiveness to sample queries` , without going into the details of random, round robin, duplicate, or something else.

Reply:
applied @cskiraly's suggestion in https://github.com/ethereum/consensus-specs/pull/3574/commits/8150f76e1c743644343371c9dab67004bd254fa7

thanks!